### ISSUE TITLE
Mapbox crashes randomly, normally in low gps areas or after hard closing app.

### ISSUE THREAD

**Issue by sillerjp** [mapbox/mapbox-events-ios#54]
```
0                                                                       testappiOS                     0x00000001039f6c10 mono_handle_native_crash + 224
2018-07-31 18:39:58.194 testappiOS[3659:1456484] critical:     1   testappiOS                     0x0000000103a0376c mono_sigsegv_signal_handler + 316
2018-07-31 18:39:58.194 testappiOS[3659:1456484] critical:     2   libsystem_platform.dylib            0x0000000183190b58 _sigtramp + 52
2018-07-31 18:39:58.194 testappiOS[3659:1456484] critical:     3   Mapbox                              0x0000000108757680 _Z45MBGLCustomGeometrySourceOptionsFromDictionaryP12NSDictionaryIP8NSStringP11objc_objectE + 180512
2018-07-31 18:39:58.194 testappiOS[3659:1456484] critical:     4   Mapbox                              0x00000001087578b8 _Z45MBGLCustomGeometrySourceOptionsFromDictionaryP12NSDictionaryIP8NSStringP11objc_objectE + 181080
2018-07-31 18:39:58.194 testappiOS[3659:1456484] critical:     5   CoreLocation                        0x0000000189d0e7b0 CLClientRetrieveData + 72260
2018-07-31 18:39:58.194 testappiOS[3659:1456484] critical:     6   CoreLocation                        0x0000000189d0e014 CLClientRetrieveData + 70312
2018-07-31 18:39:58.194 testappiOS[3659:1456484] critical:     7   CoreLocation                        0x0000000189cf79e8 CLClientInvalidate + 1240
2018-07-31 18:39:58.194 testappiOS[3659:1456484] critical:     8   CoreFoundation                      0x00000001835172bc <redacted> + 20
2018-07-31 18:39:58.194 testappiOS[3659:1456484] critical:     9   CoreFoundation                      0x0000000183516a7c <redacted> + 264
2018-07-31 18:39:58.194 testappiOS[3659:1456484] critical:     10  CoreFoundation                      0x00000001835147b0 <redacted> + 1224
2018-07-31 18:39:58.195 testappiOS[3659:1456484] critical:     11  CoreFoundation                      0x0000000183434da8 CFRunLoopRunSpecific + 552
2018-07-31 18:39:58.195 testappiOS[3659:1456484] critical:     12  GraphicsServices                    0x000000018541a020 GSEventRunModal + 100
2018-07-31 18:39:58.195 testappiOS[3659:1456484] critical:     13  UIKit                               0x000000018d454758 UIApplicationMain + 236
```

**Comment by rclee** [comment]
Hi @JPSiller 

Thanks for the report! We're actively investigating and believe this crash may be related to: https://github.com/mapbox/mapbox-events-ios/issues/50

Could you include a list of dependencies in your application CocoaPods or otherwise?

**Comment by sillerjp** [comment]
Hi rclee,

The link doesn't work but I think I know which one are you referring too. I'm actually using Xamarin and I created bindings to the map box library. The error seems to originate in the darwin framework . Would the carthage file work for your research?

**Comment by rclee** [comment]
Typo on the link, sorry. 

>Would the carthage file work for your research?

Yes, anything to tell us what your dependencies are would be sufficient. Thanks!

**Comment by sillerjp** [comment]
I noticed this happens only when I set ShowUsersLocation = true. Is this related to the merge?

**Comment by sillerjp** [comment]
What version of Mapbox GL will include this fix? Currently Mapbox is version 4.2

**Comment by rclee** [comment]
@JPSiller The next version (v4.3.0) should include this fix. If the crashes have become too much a hindrance then I recommend reverting to a version prior to v4.0.0 until the next Maps SDK release. Thank you for your patience.

**Comment by sillerjp** [comment]
Thanks!

**Comment by sillerjp** [comment]
Hi, I downloaded Mapbox 4.3.0 Beta to test if this had been fixed:

-Fixed a crash that occurred when MMELocationManager was deallocated and the delegate was reporting updates. (#12542)

I'm still getting this crash, it might be a different issue but seems related to that bug (please see stacktrack at the end of this message). To replicate it I have to do the following:

Set ShowLocation to TRUE <-- This is the most important step to reproduce, seems that if I set it to false the issue doesn't happen but of course we don't have the blue dot showing the user location which is not a valid solution to us. I've also tried to add that line to another part of the application lifecycle but it doesn't help.

Run the app, disable WIFI and just use data network
Hard Close the app
Open it again

It doesn't happen all the time, I normally have to walk around a room or go to places where I think GPS connection is bad and then I get the crash, now, the thing is, that once you get into this crash, you can keep running the app and it will keep crashing when starting, it will crash around 8 to 10 times and then at some point it recovers. 

We are getting a lot of bad reviews in the appstore related to this issue, I'm also not discarding the possibility that it could be on our end but seems like a native issue in the library. Any ideas would be helpful.
Thanks in advance

—
```
2018-08-09 09:28:05.697 CrashingAppiOS[11822:4976651] critical: 	0   CrashingAppiOS                     0x00000001052d2d30 mono_handle_native_crash + 224
2018-08-09 09:28:05.697 CrashingAppiOS[11822:4976651] critical: 	1   CrashingAppiOS                     0x00000001052df88c mono_sigsegv_signal_handler + 316
2018-08-09 09:28:05.697 CrashingAppiOS[11822:4976651] critical: 	2   libsystem_platform.dylib            0x0000000183190b58 _sigtramp + 52
2018-08-09 09:28:05.697 CrashingAppiOS[11822:4976651] critical: 	3   Mapbox                              0x0000000109da0a4c _Z45MBGLCustomGeometrySourceOptionsFromDictionaryP12NSDictionaryIP8NSStringP11objc_objectE + 171232
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	4   Mapbox                              0x0000000109da0c68 _Z45MBGLCustomGeometrySourceOptionsFromDictionaryP12NSDictionaryIP8NSStringP11objc_objectE + 171772
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	5   CoreLocation                        0x0000000189d0e7b0 CLClientRetrieveData + 72260
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	6   CoreLocation                        0x0000000189d0e014 CLClientRetrieveData + 70312
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	7   CoreLocation                        0x0000000189cf79e8 CLClientInvalidate + 1240
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	8   CoreFoundation                      0x00000001835172bc <redacted> + 20
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	9   CoreFoundation                      0x0000000183516a7c <redacted> + 264
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	10  CoreFoundation                      0x0000000183514abc <redacted> + 2004
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	11  CoreFoundation                      0x0000000183434da8 CFRunLoopRunSpecific + 552
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	12  GraphicsServices                    0x000000018541a020 GSEventRunModal + 100
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	13  UIKit                               0x000000018d454758 UIApplicationMain + 236
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	14  CrashingAppiOS                     0x0000000103120fa0 wrapper_managed_to_native_UIKit_UIApplication_UIApplicationMain_int_string___intptr_intptr + 368
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	15  CrashingAppiOS                     0x00000001030a5fbc UIKit_UIApplication_Main_string___intptr_intptr + 44
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	16  CrashingAppiOS                     0x00000001030a5f7c UIKit_UIApplication_Main_string___string_string + 172
2018-08-09 09:28:05.698 CrashingAppiOS[11822:4976651] critical: 	17  CrashingAppiOS                     0x0000000102b73f74 CrashingApp_iOS_Application_Main_string__ + 164
2018-08-09 09:28:05.699 CrashingAppiOS[11822:4976651] critical: 	18  CrashingAppiOS                     0x0000000102fde1f8 wrapper_runtime_invoke_object_runtime_invoke_dynamic_intptr_intptr_intptr_intptr + 248
2018-08-09 09:28:05.699 CrashingAppiOS[11822:4976651] critical: 	19  CrashingAppiOS                     0x00000001052e2984 mono_jit_runtime_invoke + 904
2018-08-09 09:28:05.699 CrashingAppiOS[11822:4976651] critical: 	20  CrashingAppiOS                     0x000000010535fe3c mono_runtime_invoke_checked + 148
2018-08-09 09:28:05.699 CrashingAppiOS[11822:4976651] critical: 	21  CrashingAppiOS                     0x0000000105365bb4 mono_runtime_exec_main_checked + 120
2018-08-09 09:28:05.699 CrashingAppiOS[11822:4976651] critical: 	22  CrashingAppiOS                     0x00000001052c5444 mono_jit_exec + 268
2018-08-09 09:28:05.699 CrashingAppiOS[11822:4976651] critical: 	23  CrashingAppiOS                     0x000000010544f6e8 xamarin_main + 2220
2018-08-09 09:28:05.699 CrashingAppiOS[11822:4976651] critical: 	24  CrashingAppiOS                     0x0000000102b70000 main + 96
2018-08-09 09:28:05.699 CrashingAppiOS[11822:4976651] critical: 	25  libdyld.dylib                       0x0000000182ec5fc0 <redacted> + 4
2018-08-09 09:28:05.699 CrashingAppiOS[11822:4976651] critical: 
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================
```

**Comment by rclee** [comment]
@JPSiller Sorry to hear the fix didn't work for you. We'll continue to look into this issue and keep you updated. Thanks for the detailed steps and stacktrace, that will help immensely.

**Comment by sillerjp** [comment]
Thanks @rclee 

I don't think downgrading to version 4 will be something we can do but I'll try and see if it works in that version. Any other info please let me know since we are looking forward to fixing this issue asap even if it requires using an alpha or beta Mapbox version

**Comment by sillerjp** [comment]
HI @rclee any updates on this issue? Would you be able to point me to the files referenced in the crash log to see if at least we can find a clue as to what is happening and try to do a workaround or fix on our side? We still haven't been able to fix this but we have pin pointed the issue to happen exactly after:

ShowsUserLocation = true is set 

It is like a location update cames through with bad data which crashes the logic that shows user location in the map.

Thanks

**Comment by rclee** [comment]
Hi @JPSiller we'll have a beta next week for you to try with a few crash fixes that should hopefully resolve the crash you are experiencing as well. Thanks again for your patience while we work through this problem.

**Comment by sillerjp** [comment]
Thanks! I'll keep you updated in case we find something in our side.

**Comment by ghost** [comment]
I'm eagerly waiting for this fix. We were planning a new app release and this mysterious bug appeared in the past few days out of nowhere.

**Comment by sillerjp** [comment]
Hi @sdtaheri ,

for me it fixed by removing the access token key from the info.plist and setting it directly in code when necessary.

It seems related to the Mapbox Directions library rather than the main mapbox library.

This is fixed for me now. cc @rclee

**Comment by rclee** [comment]
@JPSiller Sorry this version didn't work for you. We'll continue to look into this issue to come up with a more permanent solution. The information you've provided will help a lot in that regard. 

Glad to hear that you found a workaround!

**Comment by sillerjp** [comment]
No problem I think this is already fixed if you update the directions library but since we use Xamarin is harder for us to recreate the bindings so this workaround is pretty handy.

Thanks!

**Comment by rclee** [comment]
We believe this issue has been resolved in `4.4.1` patch release of the Maps SDK. Closing this ticket for now.
